src/test/groovy/com/pszymczyk/consul/infrastructure/ConsulWaiterTest.groovy

```diff
@@ -14,4 +18,80 @@ class ConsulWaiterTest extends Specification {
         def ex = thrown EmbeddedConsulException
         ex.message =~ "Could not start Consul*"
     }
+
+
+    def "should use token to get list of registered nodes"() {
+        when:
+        def uuid = UUID.randomUUID().toString()
+        def conf = """{
+          "datacenter": "us",
+          "acl_datacenter": "us",
+          "acl_default_policy": "deny",
+          "acl_agent_master_token": ${uuid},
+          "acl_master_token": ${uuid},
+          "acl_agent_token": ${uuid},
+          "acl_token": ${uuid},
+          "acl_enable_key_list_policy": true,
+          "acl_down_policy": "deny"
+        }"""
+
+        ConsulProcess consul = ConsulStarterBuilder.consulStarter()
+                .withToken(uuid)
+                .withCustomConfig(conf)
+                .withConsulVersion("1.0.0")
+                .build().start()
+
+        then:
+        noExceptionThrown()
```

Comment:
pls check somehow is Consul process already started here, you could use 
```
void awaitUntilServiceRegistered(String id) 
```
from ```ConsulTestWaiter``` for example, now this tests does not test anything, if I remove whole given section it will be passing all the time

Reply:
Probably the name is misleading - my bad. It refers to inner call to `ConsulWaiter::allNodesRegistered`, which can not be done if token is incorrect